hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -896,8 +897,12 @@ public AbfsInputStream openFileForRead(Path path,
         AbfsHttpOperation op = getClient().getPathStatus(relativePath, false,
             tracingContext, null).getResult();
         resourceType = getClient().checkIsDir(op) ? DIRECTORY : FILE;
-        contentLength = Long.parseLong(
-            op.getResponseHeader(HttpHeaderConfigurations.CONTENT_LENGTH));
+        String contentLengthHeader = op.getResponseHeader(HttpHeaderConfigurations.CONTENT_LENGTH);
```

Comment:
We can create a utility method extractContentLength(op.getResult()) having these checks.
Same can be used at other places as well.

Reply:
Taken